substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2784,6 +2790,19 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Apply a pending slash on a member.
+		#[pallet::call_index(23)]
+		// FIXME(ank4n): fix weight info.
+		#[pallet::weight(T::WeightInfo::set_commission_claim_permission())]
+		pub fn apply_slash(
```

Comment:
Should this be a refunded operation if the slash is successful?

Reply:
The reporter gets back part of the slash. Currently it gets `slash_reward_fraction` of the slash (which is 10% iirc). This probably is too high and I can add another perbill configuration for these reporters specifically which would be lower.